### PR TITLE
feat: added all functionalities to the common analytics services

### PR DESCRIPTION
**Pull request by mustafadincay**
added new functions to common analytics services

### CODE REVIEW

File: analytics/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,3 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="com.hms.lib.commonmobileservices.analytics">
+
+    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
+    <uses-permission android:name="android.permission.INTERNET" />
+    <uses-permission android:name="android.permission.WAKE_LOCK" />
+
+    <meta-data
```
Comment (omerakkus) [inline analytics/src/main/AndroidManifest.xml:9]: you do not have to add this line in AndroidManifest.xml file. Please remove this code line.

File: analytics/src/main/java/com/hms/lib/commonmobileservices/analytics/AnalyticsExt.kt [hunk 2]
```diff
@@ -30,7 +30,7 @@ inline fun <reified T> CommonAnalytics.saveSingleData(
         is Double -> bundle.putDouble(paramKeyGen, data)
         is Float -> bundle.putFloat(paramKeyGen, data)
         else -> {
-            Log.d("ShoppingAnalytics", "Invalid type")
```
Comment (oguzozsoy) [inline analytics/src/main/java/com/hms/lib/commonmobileservices/analytics/AnalyticsExt.kt:33]: I would suggest not to change log type

File: analytics/src/main/java/com/hms/lib/commonmobileservices/analytics/AnalyticsExt.kt [hunk 3]
```diff
@@ -49,11 +49,10 @@ fun CommonAnalytics.saveData(eventKey: String, vararg keyDataPairs: Any) {
                 is Double -> bundle.putDouble(value as String, data)
                 is Float -> bundle.putFloat(value as String, data)
                 else -> {
-                    Log.d("ShoppingAnalytics", "Invalid type")
```
Comment (oguzozsoy) [inline analytics/src/main/java/com/hms/lib/commonmobileservices/analytics/AnalyticsExt.kt:52]: I would suggest not to change log type

**Review by omerakkus — CHANGES_REQUESTED** [review]
I would suggest to remove install_channel code line in AndroidManifest file.

File: analytics/src/main/java/com/hms/lib/commonmobileservices/analytics/GMSAnalyticsImpl.kt [hunk 4]
```diff
@@ -13,12 +13,51 @@
...
+    }
+
+    override fun addDefaultEventParams(params: Bundle) {
+        analytics.setDefaultEventParameters(params)
+    }
+
+    override fun getAAID(): Work<String> {
+        val worker: Work<String> = Work()
+        analytics.appInstanceId
+            .addOnSuccessListener { worker.onSuccess(String()) }
```
Comment (oguzozsoy) [inline analytics/src/main/java/com/hms/lib/commonmobileservices/analytics/GMSAnalyticsImpl.kt]: I think you should return appInstanceId at the onSuccess callback

File: analytics/src/main/java/com/hms/lib/commonmobileservices/analytics/HMSAnalyticsImpl.kt [hunk 5]
```diff
@@ -15,19 +15,59 @@ package com.hms.lib.commonmobileservices.analytics
...
 
+    override fun addDefaultEventParams(params: Bundle) {
+        hmsAnalytics.addDefaultEventParams(params)
+    }
+
+    override fun getAAID(): Work<String> {
+        val worker: Work<String> = Work()
+        hmsAnalytics.aaid
+            .addOnSuccessListener {
+                worker.onSuccess(String())
```
Comment (oguzozsoy) [inline analytics/src/main/java/com/hms/lib/commonmobileservices/analytics/HMSAnalyticsImpl.kt]: I think you should return aaid at the onSuccess

**Review by oguzozsoy — CHANGES_REQUESTED** [review]
Please update according to our comments.

**Review by oguzozsoy — APPROVED** [review]
it looks good.

**Review by omerakkus — APPROVED** [review]
ok

**Pull request merged**